### PR TITLE
Modernize functional tests II

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

**Comment by mdellweg** [comment]
Let's call this another milestone. The `functional/cli` subdirectory is now pulp-smash free.

File: pulp_ansible/tests/functional/conftest.py [hunk 1]
```diff
@@ -230,36 +229,49 @@ def _ansible_distribution_factory(repository=None, **kwargs):
...
-def ansible_role_remote_factory(ansible_remote_role_api_client, gen_object_with_cleanup):
+def ansible_role_remote_factory(
+    bindings_cfg, ansible_remote_role_api_client, gen_object_with_cleanup
+):
     """A factory to generate an Ansible Collection Remote with auto-cleanup."""
 
     def _ansible_role_remote_factory(**kwargs):
         kwargs.setdefault("name", str(uuid.uuid4()))
         kwargs.setdefault("url", ANSIBLE_GALAXY_URL)
+        if kwargs.pop("include_pulp_auth", False):
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/conftest.py:240]: This is a neat feature!

File: pulp_ansible/tests/functional/cli/test_collection_upload.py [hunk 2]
```diff
@@ -1,153 +1,80 @@
...
+    wait_tasks(repository.pulp_href)
+
+    repository = ansible_repo_api_client.read(repository.pulp_href)
+    assert repository.latest_version_href.endswith("/1/")
+    collections = galaxy_v3_collections_api_client.list(distribution.base_path)
+    assert collections.meta.count == 1  # We uploaded 1 collection
+    version = galaxy_v3_collection_versions_api_client.read(
+        collection_name, "pulp", distribution.base_path, "1.0.0"
+    )
+    assert version.requires_ansible == ">=2.9"
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/cli/test_collection_upload.py:80]: Nice job combining the two tests into one.

File: pulp_ansible/tests/functional/cli/test_role_install.py [hunk 3]
```diff
@@ -2,73 +2,44 @@
...
+    assert not path.exists(directory), "Directory {} already exists".format(directory)
+    subprocess.run(
+        (
+            "ansible-galaxy",
+            "role",
+            "install",
+            ANSIBLE_ELASTIC_ROLE,
+            "-c",
+            # "-s",
+            # distribution.client_url,
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/cli/test_role_install.py:38]: With the `ansible_dir_factory` these two options are no longer needed, right?

**Review by gerrod3 — COMMENTED** [review]
What else is still using pulp_smash? I see there are still some imports in the utils file, but are any tests still reliant on pulp_smash?

File: pulp_ansible/tests/functional/cli/test_role_install.py (hunk 3 above)
Comment (mdellweg) [reply pulp_ansible/tests/functional/cli/test_role_install.py:38]: Oh... They even break authorization at that point.

**Review by gerrod3 — APPROVED** [review]
Another milestone achieved! Soon pulp-smash will be no more.

**Pull request merged**